editor/editor_properties.cpp

```diff
@@ -730,265 +730,294 @@ EditorPropertyFlags::EditorPropertyFlags() {
 
 ///////////////////// LAYERS /////////////////////////
 
-class EditorPropertyLayersGrid : public Control {
-	GDCLASS(EditorPropertyLayersGrid, Control);
-
-private:
-	Vector<Rect2> flag_rects;
-	Rect2 expand_rect;
-	bool expand_hovered = false;
-	bool expanded = false;
-	int expansion_rows = 0;
-	int hovered_index = -1;
-	bool read_only = false;
-
-	Size2 get_grid_size() const {
-		Ref<Font> font = get_theme_font(SNAME("font"), SNAME("Label"));
-		int font_size = get_theme_font_size(SNAME("font_size"), SNAME("Label"));
-		return Vector2(0, font->get_height(font_size) * 3);
+void EditorPropertyLayersGrid::_rename_pressed(int p_menu) {
+	// Show rename popup for active layer.
+	if (renamed_layer_index == -1) {
+		return;
 	}
+	String name = names[renamed_layer_index];
+	rename_dialog->set_title(TTR("Renaming layer " + itos(renamed_layer_index + 1) + ":"));
+	rename_dialog_text->set_text(name);
+	rename_dialog_text->select(0, name.length());
+	rename_dialog->popup_centered(Size2(300, 80) * EDSCALE);
+	rename_dialog_text->grab_focus();
+}
+void EditorPropertyLayersGrid::_rename_operation_confirm() {
+	String new_name = rename_dialog_text->get_text().strip_edges();
+	if (new_name.length() == 0) {
+		EditorNode::get_singleton()->show_warning(TTR("No name provided."));
+		return;
+	} else if (new_name.find("/") != -1 || new_name.find("\\") != -1 || new_name.find(":") != -1) {
+		EditorNode::get_singleton()->show_warning(TTR("Name contains invalid characters."));
+		return;
+	}
+	names.set(renamed_layer_index, new_name);
+	tooltips.set(renamed_layer_index, new_name + "\n" + vformat(TTR("Bit %d, value %d"), renamed_layer_index, 1 << renamed_layer_index));
+	emit_signal(SNAME("rename_confirmed"), renamed_layer_index, new_name);
+}
+Size2 EditorPropertyLayersGrid::get_grid_size() const {
+	Ref<Font> font = get_theme_font(SNAME("font"), SNAME("Label"));
+	int font_size = get_theme_font_size(SNAME("font_size"), SNAME("Label"));
+	return Vector2(0, font->get_height(font_size) * 3);
+}
+EditorPropertyLayersGrid::EditorPropertyLayersGrid() {
```

Comment:
```suggestion
	rename_dialog_text->grab_focus();
}

void EditorPropertyLayersGrid::_rename_operation_confirm() {
	String new_name = rename_dialog_text->get_text().strip_edges();
	if (new_name.length() == 0) {
		EditorNode::get_singleton()->show_warning(TTR("No name provided."));
		return;
	} else if (new_name.find("/") != -1 || new_name.find("\\") != -1 || new_name.find(":") != -1) {
		EditorNode::get_singleton()->show_warning(TTR("Name contains invalid characters."));
		return;
	}
	names.set(renamed_layer_index, new_name);
	tooltips.set(renamed_layer_index, new_name + "\n" + vformat(TTR("Bit %d, value %d"), renamed_layer_index, 1 << renamed_layer_index));
	emit_signal(SNAME("rename_confirmed"), renamed_layer_index, new_name);
}

Size2 EditorPropertyLayersGrid::get_grid_size() const {
	Ref<Font> font = get_theme_font(SNAME("font"), SNAME("Label"));
	int font_size = get_theme_font_size(SNAME("font_size"), SNAME("Label"));
	return Vector2(0, font->get_height(font_size) * 3);
}

EditorPropertyLayersGrid::EditorPropertyLayersGrid() {
```

Reply:
Not very visible on the diff - I'm adding a line of separation between all method implementations.

Reply:
Thanks. Done.